tidb-binlog/pump_client/client.go

```diff
@@ -241,14 +242,19 @@ func (c *PumpsClient) getPumpStatus(pctx context.Context) (revision int64, err e
 
 // WriteBinlog writes binlog to a situable pump. Tips: will never return error for commit/rollback binlog.
 func (c *PumpsClient) WriteBinlog(binlog *pb.Binlog) error {
+	c.RLock()
+	pumpNum := len(c.Pumps.AvaliablePumps)
+	selector := c.Selector
```

Comment:
why putting it into a lock scope

Reply:
Ok，I think you can put lock in selector, and provide an update function. it's better

Reply:
when change strategy will create a new selector. And selector is a member of pump client. so I think lock pump client seems better